internal/provider/inventory_data_source.go

```diff
@@ -118,17 +151,63 @@ func (d *InventoryDataSource) Configure(_ context.Context, req datasource.Config
 	d.client = client
 }
 
-// inventoryDataSourceModel maps the data source schema data.
-type InventoryDataSourceModel struct {
-	Id           types.Int64                      `tfsdk:"id"`
-	Organization types.Int64                      `tfsdk:"organization"`
-	Url          types.String                     `tfsdk:"url"`
-	Name         types.String                     `tfsdk:"name"`
-	Description  types.String                     `tfsdk:"description"`
-	Variables    customtypes.AAPCustomStringValue `tfsdk:"variables"`
+func (d *InventoryDataSource) ConfigValidators(_ context.Context) []datasource.ConfigValidator {
+	// You have at least an id or a name + organization_name pair
+	return []datasource.ConfigValidator{
+		datasourcevalidator.Any(
+			datasourcevalidator.AtLeastOneOf(
+				tfpath.MatchRoot("id")),
+			datasourcevalidator.RequiredTogether(
+				tfpath.MatchRoot("name"),
+				tfpath.MatchRoot("organization_name")),
+		),
+	}
+}
+
+func (d *InventoryDataSource) ValidateConfig(ctx context.Context, req provider.ValidateConfigRequest, resp *provider.ValidateConfigResponse) {
+	var data InventoryDataSourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if IsValueProvided(data.Id) {
+		return
+	}
+
+	if IsValueProvided(data.Name) && IsValueProvided(data.OrganizationName) {
+		return
+	}
+
+	if IsValueNotProvided(data.Id) && IsValueNotProvided(data.Name) {
+		resp.Diagnostics.AddAttributeWarning(
+			tfpath.Root("id"),
+			"Missing Atribute Configuration",
+			"Expected either [id] or [name + organization_name] pair",
+		)
+	}
+
+	if IsValueProvided(data.Name) && IsValueNotProvided(data.OrganizationName) {
```

Comment:
Which is easier to read, two functions with very similar names but complete opposite returns, or, `!IsValueProvided()`

Reply:
No need for a poll, its not that important, and ok the way it is. 

I personally find the `!` more readable but its grand the way it is